nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateRecord.java

```diff
@@ -0,0 +1,649 @@
+            Object returnObject = fakerMethodHolder.getMethod().invoke(fakerMethodHolder.getMethodObject());
+            return returnObject;
+        } catch (InvocationTargetException | IllegalAccessException e) {
+            throw new ProcessException(type + " is not a valid value");
+        }
+    }
+
+    // This method overrides the default String type for certain Faker datatypes for more user-friendly values (such as a Double for latitude/longitude)
+    private DataType getDataType(final String type) {
+
+        if (FT_FUTURE_DATE.getValue().equals(type)
+                || FT_PAST_DATE.getValue().equals(type)
+                || FT_BIRTHDAY.getValue().equals(type)
+        ) {
+            return RecordFieldType.DATE.getDataType();
+        }
+        if (FT_LATITUDE_ALLOWABLE_VALUE_NAME.equals(type)
+                || FT_LONGITUDE_ALLOWABLE_VALUE_NAME.equals(type)) {
+            return RecordFieldType.DOUBLE.getDataType("%.8g");
+        }
+        if (FT_NUMBER.getValue().equals(type)) {
+            return RecordFieldType.INT.getDataType();
+        }
+        if (FT_BOOL.getValue().equals(type)) {
+            return RecordFieldType.BOOLEAN.getDataType();
+        }
+        return RecordFieldType.STRING.getDataType();
+    }
+
+    private Object generateValueFromRecordField(RecordField recordField, Faker faker, int nullPercentage) {
+        if (recordField.isNullable() && faker.number().numberBetween(0, 100) <= nullPercentage) {
+            return null;
+        }
+        switch (recordField.getDataType().getFieldType()) {
+            case BIGINT:
+                return new BigInteger(String.valueOf(faker.number().numberBetween(Long.MIN_VALUE, Long.MAX_VALUE)));
+            case BOOLEAN:
+                return getFakeData("Bool.bool", faker);
+            case BYTE:
+                return faker.number().numberBetween(Byte.MIN_VALUE, Byte.MAX_VALUE);
+            case CHAR:
+                return (char) faker.number().numberBetween(Character.MIN_VALUE, Character.MAX_VALUE);
+            case DATE:
+                return getFakeData(FT_PAST_DATE.getValue(), faker);
+            case DECIMAL:
+            case DOUBLE:
+            case FLOAT:
+                return faker.number().randomDouble(((DecimalDataType) recordField.getDataType()).getScale(), Long.MIN_VALUE, Long.MAX_VALUE);
+            case INT:
+                return faker.number().numberBetween(Integer.MIN_VALUE, Integer.MAX_VALUE);
+            case LONG:
+                return faker.number().numberBetween(Long.MIN_VALUE, Long.MAX_VALUE);
+            case SHORT:
+                return faker.number().numberBetween(Short.MIN_VALUE, Short.MAX_VALUE);
+            case ENUM:
+                List<String> enums = ((EnumDataType) recordField.getDataType()).getEnums();
+                return enums.get(faker.number().numberBetween(0, enums.size() - 1));
+            case TIME:
+                DateFormat df = new SimpleDateFormat("HH:mm:ss");
```

Comment:
`SimpleDateFormat` usage should be replaced with `DateTimeFormatter`, which is thread-safe and can be declared as a member variable.